src/layout/Summary2/SummaryComponent2/SummaryComponent2.tsx

```diff
@@ -12,52 +12,40 @@ export interface ISummaryComponent2 {
 }
 
 export function _SummaryComponent2({ summaryNode }: ISummaryComponent2) {
-  if (summaryNode.item.whatToRender.type === 'layoutSet') {
+  if (!summaryNode.item.target && !summaryNode.item.taskId) {
+    return <LayoutSetSummary />;
+  }
+
+  if (summaryNode.item.taskId) {
```

Comment:
Ohh, so you can point to one page or component in a different task? :star_struck: But can you point to just "the whole task" as well, aka all pages in that task? If we make app developers have to add multiple Summary2 components to point to different pages in the same (previous) task, we'll also force ourselves to generate state (like nodes) independently for each Summary2 component. When the hierarchy rewrite is merged, I suspect that will add lots of overhead and duplicated state in memory.

One alternative is to try to deduplicate `<FormProvider>` across multiple Summary2 components with the same task, but it will complicate things quite a lot if we have to store "state for a task" in an external store disconnected from the render tree..

Anyway - one more thing I thought when seeing this - shouldn't `taskId` also be a property of `target`? It's very related to the target, at least.

Reply:
Moved the taskId to target :) I see your point with regards to multiple states. In light of our discussions about navigation vs tasks, this feature might become less important though, so I dont think we should put too much effort into it at this point, but we can mention in the docs that we advice against putting several summary components that render previous tasks on the same page for example :)